tests/semanticdb/expect/Classes.expect.scala

```diff
@@ -11,7 +11,7 @@ case class C4/*<-classes::C4#*/(x/*<-classes::C4#x.*/: Int/*->scala::Int#*/)
 object C4/*<-classes::C4.*/
 
 object M/*<-classes::M.*/ {
-  implicit class C5/*<-classes::M.C5#*/(x/*<-classes::M.C5#x.*/: Int/*->scala::Int#*/)
+  /*->classes::M.C5().(x)*/implicit class C5/*<-classes::M.C5#*//*<-classes::M.C5().*/(x/*<-classes::M.C5#x.*//*<-classes::M.C5().(x)*/: Int/*->scala::Int#*/)
```

Comment:
This should not be added to semanticdb occurrences, they are synthetic for the source code point of view :thinking: 

For example ` /*->classes::M.C5().(x)*/` doesn't really correspond to any symbol, it's at the start of the line, where no symbol is declared.

Reply:
this can be solved by adding to `excludeDef` and `excludeChildren` a rule for implicit class conversions:
```scala
extension (sym: Symbol) // addition within `Scala3.SymbolOps` given
  def isImplicitClassConvertor(using Context) =
    sym.isAllOf(Implicit|Method) && sym.isCoDefinedGiven(sym.info.finalResultType.classSymbol)

...

private def excludeDef(sym: Symbol)(using Context): Boolean =
  !sym.exists
  || ...
  || sym.isImplicitClassConvertor
  || ...
  || excludeDefOrUse(sym)

...

private def excludeChildren(sym: Symbol)(using Context): Boolean =
  !sym.exists
  || ...
  || sym.isImplicitClassConvertor
```

Reply:
@bishabosha Can you take this over and do the exclude changes please?

Reply:
It looks like it has been fixed in https://github.com/lampepfl/dotty/pull/12885 (specifically https://github.com/lampepfl/dotty/pull/12885/commits/fef5be020c9e7c02446772140910b9e6a03aeaa8 ), those occurrences will disappear once we rebase onto master.